containers/DvmState.ts

```diff
@@ -0,0 +1,140 @@
+
+  // Because apollo caches results of queries, we will poll/refresh this query periodically.
+  // We set the poll interval to a very slow 5 seconds for now since the position states
+  // are not expected to change much.
+  // Source: https://www.apollographql.com/docs/react/data/queries/#polling
+  const { loading, error, data } = useQuery(PRICE_REQUESTS, {
+    context: { clientName: "UMA" },
+    pollInterval: 5000,
+  });
+
+  // get state from EMP
+  const queryState = async () => {
+    if (
+      dvm !== null &&
+      priceIdentifier !== null &&
+      expirationTimestamp !== null
+    ) {
+      const res = await Promise.all([
+        dvm.getPriceRequestStatuses([
+          {
+            identifier: priceIdentifier.toString(),
+            time: expirationTimestamp.toNumber(),
+          },
+        ]),
+      ]);
+
+      const hasPrice = res[0][0].status === PRICE_REQUEST_STATUSES.RESOLVED;
+
+      let resolvedPrice = null;
+
+      // Fetch price from graph if available.
+      if (error) {
+        console.error(`Apollo client failed to fetch graph data:`, error);
+      } else if (hasPrice && !loading && data) {
+        // Implementation 1: Use subgraph data. This works for mainnet but not Kovan since the UMA subgraph does not currently index Kovan data.
+        const matchingPriceRequest = data.priceRequests.find(
+          (request: PriceRequestQuery) => {
+            return (
+              request.identifier.id === hexToUtf8(priceIdentifier) &&
+              request.time === expirationTimestamp.toString()
+            );
+          }
+        );
+        if (matchingPriceRequest && matchingPriceRequest.isResolved) {
+          resolvedPrice = parseFloat(
+            fromWei(matchingPriceRequest.price.toString())
+          );
+        }
+
+        // // Implementation 2: Call contract. This works for Kovan/MockOracle and is therefore useful for testing, but not mainnet since `getPrice` must be called with the overrideL `{from: emp.address}`.
+        // try {
+        //   const postResolutionRes = await Promise.all([
+        //     // `getPrice()` will fail on mainnet because the deployed `Voting.getPrice()` is required to be called from a registered contract, such as this EMP!
+        //     // TODO: Figure out a way to spoof the `from` parameter in this contract call and set to `emp.address`.
+        //     dvm.getPrice(
+        //       priceIdentifier.toString(),
+        //       expirationTimestamp.toNumber()
+        //     ),
+        //   ]);
+        //   resolvedPrice = parseFloat(fromWei(postResolutionRes.toString()));
+        // } catch (err) {
+        //   console.error(`getPrice failed:`, err);
+        // }
+      }
+
+      const newState: ContractState = {
+        hasEmpPrice: hasPrice,
+        resolvedPrice: resolvedPrice,
+      };
+
+      setState(newState);
+    }
+  };
+
+  // get state on setting of contract
+  useEffect(() => {
+    setState(initState);
```

Comment:
Do you want to re-set the state like this within the `useEffect`? I _think_ this will result in the state having a short discontinuity to `null` between polls. The result of this is that while the state is updated in the `queryState` function anything that is dependent on this state will flicker to the `null` state and then back to the populated state when the data has loaded.

I think it's fine to just remove this line as the state is populated to the initState when it's created and updated to newState on each loop.

I came across this behavior while adding the final fee to the Dispute params in the front end. I forked off this branch and added the `finalFee` to the dvm's `ContractState`. This is in [this](https://github.com/UMAprotocol/emp-tools/pull/103) PR. Once your PR is merged into master I'll point that PR to master as well.

Reply:
Ah you're right! I was wondering why SettleExpired was flickering haha